openfermioncirq/gates/three_qubit_gates_test.py

```diff
@@ -180,3 +182,26 @@ def test_three_qubit_gate_text_diagrams():
       │           │
 c: ───XXYY^-0.5───#2^-0.5───
 """)
+
+
+def test_combined_cxxyy():
+    ofc.testing.assert_eigengate_implements_consistent_protocols(
+        ofc.CombinedCXXYYPowGate)
+
+@pytest.mark.parametrize('exponent,control',
+    itertools.product(
+        [0, 1, -1, 0.25, -0.5, 0.1],
+        [0, 1, 2]))
+def test_combined_cxxyy_consistency(exponent, control):
+    weights = tuple(np.eye(1, 3, control)[0])
+    general_gate  = ofc.CombinedCXXYYPowGate(weights, exponent=exponent)
+    general_unitary = cirq.unitary(general_gate)
+
+    indices = np.dot(
+            list(itertools.product((0, 1), repeat=3)),
+            (2 ** np.roll(np.arange(3), -control))[::-1])
+    special_gate = ofc.CXXYYPowGate(exponent=exponent)
+    special_unitary = (
+            cirq.unitary(special_gate)[indices[:, np.newaxis], indices])
+
+    assert np.allclose(general_unitary, special_unitary)
```

Comment:
Add a test that the matrix of the gate matches the matrix specified in the documentation.

Reply:
Done, but tests failing because of recent changes to Cirq. I addressed the `QubitId` to `Qid` change, but would need to look into how to address the changes to the simulator interface. Is anyone working on that?

Reply:
Actually, tests are failing _because_ you changed QubitId to Qid. Travis uses a specific dev version of Cirq (see the requirements.txt).

Reply:
Sorry, yes, that's technically correct; I should have been clearer. I used some things from a more recent version of Cirq than is in OFC's requirements file, so other tests fail even with the latter. I could work around those, but I figured it'd be better to get closer to in sync with Cirq master.